admin/bin/qserv-parallel-cmd.py

```diff
@@ -0,0 +1,192 @@
+                           'default: %(default)s.')
+        group.add_argument('-R', '--run-dir', dest='run_dir',
+                           default=None,
+                           help='absolute path to the directory where the command is runned'
+                           )
+        group.add_argument('-s', '--sudo-user', dest='sudo_user',
+                           default=None,
+                           help='sudo to this user account before launching command'
+                           )
+        group.add_argument('-p', '--sudo-password', dest='sudo_password',
+                           default=None,
+                           help='sudo-user password, WARNING: insecure, prefer'
+                           'password-less sudo'
+                           )
+
+        # parse all arguments
+        self.args = parser.parse_args()
+
+        verbosity = len(self.args.verbose)
+        if verbosity != 0:
+            levels = {0: logging.WARNING, 1: logging.INFO, 2: logging.DEBUG}
+            simple_format = "[%(levelname)s] %(name)s: %(message)s"
+            level = levels.get(verbosity, logging.DEBUG)
+            logging.basicConfig(format=simple_format, level=level)
+        else:
+            # if -v(vv) option isn't used then
+            # switch to global configuration file for logging
+            lsst.qserv.admin.logger.setup_logging(self.args.log_conf)
+
+        # instantiate nodes
+        if self.args.hosts:
+            hosts = self.args.hosts
+        else:
+            hosts = [self.args.host_tpl.format(n)
+                     for n in range(self.args.node_first,
+                                    self.args.node_last)]
+
+        nodes = [NodeAdmin(host=h,
+                           runDir=self.args.run_dir,
+                           kerberos=self.args.kerberos,
+                           ssh_user=self.args.user)
+                 for h in hosts]
+
+        self.nodePool = NodePool(nodes)
+
+    def run(self):
+        """
+        Run a command on a set of nodes  based on parameters defined in
+        constructor.
+        """
+        params = {'command': self.args.command}
+        cmd_tpl = '{command}'
+        if self.args.sudo_user:
+            params['sudo_user'] = self.args.sudo_user
+            params['stdin_opt'] = ''
+            cmd_tpl = 'sudo {stdin_opt} -u {sudo_user} {command}'
+            if self.args.sudo_password is not None:
+                params['stdin_opt'] = '-kS'
+                params['sudo_password'] = self.args.sudo_password
+                cmd_tpl = "/bin/echo '{sudo_password}' | " + cmd_tpl
+
+        cmd = cmd_tpl.format(**params)
+        failed = self.nodePool.execParallel(cmd, self.args.stdin)
+        _LOG.info("Node in error: '%s'", failed)
+        return failed
+
+
+if __name__ == "__main__":
+    try:
+        parallelCmd = ParallelCmd()
+        r = parallelCmd.run()
+        sys.exit(r)
```

Comment:
If you have more than 127 failed nodes then it is a potential issue, return code above 127 are "undefined behavior".